Buildings/Controls/OBC/CDL/Discrete/MovingMean.mo

```diff
@@ -0,0 +1,103 @@
+within Buildings.Controls.OBC.CDL.Discrete;
+block MovingMean "Discrete moving mean of a sampled input signal"
+  parameter Integer n(min=2) "Number of samples over which the input is averaged";
+  parameter Modelica.SIunits.Time samplePeriod(min=1E-3)
+      "Sampling period of component";
+
+  Interfaces.RealInput u "Continuous input signal"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}})));
+  Interfaces.RealOutput y "Discrete averaged signal"
+    annotation (Placement(transformation(extent={{100,-10},{120,10}}),
+        iconTransformation(extent={{100,-10},{120,10}})));
+
+protected
+  parameter Modelica.SIunits.Time t0(fixed=false) "First sample time instant";
+  Boolean sampleTrigger "Trigger samples at each sampling instant";
+  Integer iSample(start=0, fixed=true) "Sample numbering in the simulation";
+  Integer counter(start=0, fixed=true)
+      "Number of samples used for averaging calculation";
+  Integer index(start=0, fixed=true) "Index of the vector ySample";
+  Real ySample[n](start=vector(zeros(n,1)), fixed=true)
+      "Vector of samples to be averaged";
+
+initial algorithm
+  t0 := time;
+  y := 0;
```

Comment:
Better use `y := u` here. Otherwise, you could get `y` to be outside of the range of the input `u`.